diff --git a/diag/mfg/diag_qa/diag_mtp_reload.py b/diag/mfg/diag_qa/diag_mtp_reload.py
index f5959a9a..1abb420c 100755
--- a/diag/mfg/diag_qa/diag_mtp_reload.py
+++ b/diag/mfg/diag_qa/diag_mtp_reload.py
@@ -169,8 +169,9 @@ def main():
         # init the nic diag environment
         for slot in range(MTP_Const.MTP_SLOT_NUM):
             if nic_prsnt_list[slot]:
-                mtp_mgmt_ctrl.mtp_nic_diag_init(slot)
+                mtp_mgmt_ctrl.mtp_nic_mini_init(slot)
                 mtp_mgmt_ctrl.mtp_mgmt_set_nic_diag_boot(slot)
+                mtp_mgmt_ctrl.mtp_power_off_single_nic(slot)
 
     mtp_mgmt_ctrl.mtp_enter_user_ctrl()
 
diff --git a/diag/mfg/mtp_regression/mtp_diag_regression.py b/diag/mfg/mtp_regression/mtp_diag_regression.py
index b236772b..921702e3 100755
--- a/diag/mfg/mtp_regression/mtp_diag_regression.py
+++ b/diag/mfg/mtp_regression/mtp_diag_regression.py
@@ -449,17 +449,9 @@ def main():
         mtp_mgmt_ctrl.cli_log_inf("Set Default Boot up Image to SW", level=0)
         mtp_mgmt_ctrl.mtp_power_off_nic()
         mtp_mgmt_ctrl.mtp_power_on_nic()
-
         for slot in naples100_nic_list:
             mtp_mgmt_ctrl.mtp_nic_mini_init(slot)
             mtp_mgmt_ctrl.mtp_mgmt_set_nic_sw_boot(slot)
-
-        mtp_mgmt_ctrl.mtp_power_off_nic()
-        mtp_mgmt_ctrl.mtp_power_on_nic()
-
-        for slot in naples100_nic_list:
-            mtp_mgmt_ctrl.mtp_nic_mini_init(slot)
-            mtp_mgmt_ctrl.mtp_mgmt_verify_nic_sw_boot(slot)
         mtp_mgmt_ctrl.cli_log_inf("Set Default Boot up Image to SW Complete\n", level=0)
 
     for nic_key, nic_sn in zip(fail_nic_list, fail_sn_list):
